packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -722,188 +572,156 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
+        require(managerBuyPrice >= sellerPrice, "Manager Buy Price should be atleast Seller Price");
+        bytes32 structHash = keccak256(
+            abi.encode(
+                _CRAB_BALANCE_TYPEHASH,
+                _order.bidId,
+                _order.trader,
+                _order.traderToken,
+                _order.traderAmount,
+                _order.managerToken,
+                _order.managerAmount,
+                _useNonce(_order.trader)
+            )
+        );
 
-            uint256 remainingEth = _buyerAmount.sub(_ethToBuy);
+        bytes32 hash = _hashTypedDataV4(structHash);
+        address offerSigner = ECDSA.recover(hash, _order.v, _order.r, _order.s);
+        require(offerSigner == _order.trader, "Invalid offer signature");
 
-            if (remainingEth > 0) {
-                payable(_buyer).sendValue(remainingEth);
-            }
+        //adjust managerAmount and TraderAmount for partial fills
+        if (managerSellAmount < _order.managerAmount) {
+            _order.managerAmount = managerSellAmount;
         }
+        //adjust if manager is giving better price
+        _order.traderAmount = _order.managerAmount.mul(1e18).div(managerBuyPrice);
 
-        emit ExecuteSellAuction(_buyer, _wSqueethToSell, _ethToBuy, _isHedgingOnUniswap);
-    }
+        IERC20(_order.traderToken).transferFrom(_order.trader, address(this), _order.traderAmount);
 
-    /**
-     * @notice execute buy auction based on the parameters calculated
-     * @dev if _isHedgingOnUniswap, ETH proceeds are not sent to seller
-     * @param _seller seller address
-     * @param _wSqueethToBuy wSqueeth amount to buy
-     * @param _ethToSell ETH amount to sell
-     * @param _isHedgingOnUniswap true if arbing with uniswap price
-     */
-    function _executeBuyAuction(
-        address _seller,
-        uint256 _wSqueethToBuy,
-        uint256 _ethToSell,
-        bool _isHedgingOnUniswap
-    ) internal {
-        _burnWPowerPerp(_seller, _wSqueethToBuy, _ethToSell, _isHedgingOnUniswap);
-
-        if (!_isHedgingOnUniswap) {
-            payable(_seller).sendValue(_ethToSell);
+        // if the trader is selling WETH to us i.e if we are selling oSQTH
+        if (_order.traderToken == weth) {
+            IWETH9(weth).withdraw(IWETH9(weth).balanceOf(address(this)));
+            // if last param is false, transfer happens again
+            _mintWPowerPerp(_order.trader, _order.managerAmount, _order.traderAmount, true);
+            priceAtLastHedge = _order.traderAmount.mul(1e18).div(_order.managerAmount);
+        } else {
+            // oSQTH in, WETH out
+            _burnWPowerPerp(_order.trader, _order.traderAmount, _order.managerAmount, true);
+            //wrap it
+            IWETH9(weth).deposit{value: _order.managerAmount}();
+            // if last param is false, transfer happens again
+            priceAtLastHedge = _order.managerAmount.mul(1e18).div(_order.traderAmount);
```

Comment:
I think this has to be swapped.

priceAtLastHedge is stored as ETH per 1 oSQTH in contract, and this is oSQTH per ETH, correct?